.github/workflows/codeql.yml

```diff
@@ -0,0 +1,114 @@
+###
+# @format
+# -----
+# Project: zenstack-monorepo
+# File: codeql.yml
+# Path: \.github\workflows\codeql.yml
+# Created Date: Monday, February 19th 2024
+# Author: Jonathan Stevens, jonathan@resnovas.com
+# Github: https://github.com/TGTGamer
+# -----
+# Contributing: Please read through our contributing guidelines.
+# Included are directions for opening issues, coding standards,
+# and notes on development. These can be found at
+# https://github.com/zenstack-monorepo/blob/develop/CONTRIBUTING.md
+# -----
+# Code of Conduct: This project abides by the Contributor Covenant, v2.0
+# Please interact in ways that contribute to an open, welcoming, diverse,
+# inclusive, and healthy community. Our Code of Conduct can be found at
+# https://github.com/zenstack-monorepo/blob/develop/CODE_OF_CONDUCT.md
+# -----
+# Copyright (c) 2024 ZenstackHQ - All Rights Reserved
+# LICENSE: MIT License (MIT)
+# -----
+# This program has been provided under confidence of the copyright holder and
+# is licensed for copying, distribution and modification under the terms
+# of the MIT License (MIT) published as the License,
+# or (at your option) any later version of this license.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# MIT License for more details.
+# You should have received a copy of the MIT License
+# along with this program. If not, please write to: jonathan@resnovas.com,
+# or see https://opensource.org/licenses/MIT
+# -----
+# DELETING THIS NOTICE AUTOMATICALLY VOIDS YOUR LICENSE
+###
+
+name: Security - CodeQL
+
+on:
+  merge_group:
+  push:
+    branches: 
+      - main
+      - develop
+      - release/*
+      - v2
+  pull_request:
+    branches: 
+      - main
+      - develop
+      - release/*
+      - v2
+  schedule:
+    - cron: "0 0 * * 1"
+
+permissions:
+  contents: read
+  
+
+jobs:
+  analyze:
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+    name: Analyze
+    runs-on: ubuntu-latest
+    
+    strategy:
+      fail-fast: false
+      matrix:
+        language: ["javascript", "typescript"]
+        # CodeQL supports [ $supported-codeql-languages ]
+        # Learn more about CodeQL language support at https://aka.ms/codeql-docs/language-support
+
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@eb238b55efaa70779f274895e782ed17c84f2895 # v2.6.1
```

Comment:
Just curious: is it a recommended practice to pin actions to a sha?

Reply:
Depends who you discuss with. Most people advise to pin to either version or SHA. I use SHA as that's the step-security recommendation, but it has it's drawbacks in terms of updates and patches. 

Additionally, if you have dependabot checking for updates, in theory it should bump SHA as well, but I'm yet to fully test that :)

Reply:
Got it. Thank you for the explanation.